packages/@aws-cdk/custom-resource-handlers/lib/aws-stepfunctions-tasks/cross-region-aws-sdk-handler/index.ts

```diff
@@ -78,7 +78,20 @@ export const handler = async (event: Event) => {
     const command = new Command(event.parameters ?? {});
     const res = await client.send(command);
 
-    return res;
+    // Special handling for Lambda invoke responses
+    if (res.Payload && (res.Payload instanceof Uint8Array || res.Payload instanceof Buffer)) {
```

Comment:
Just a suggestion: Instead of looking at the fixed path, why not search for the Buffer element recursively in the JSON?

There are more APIs that return Buffer or Uint8Array, such as Bedrock InvokeModel, DynamoDB GetItem, KMS GenerateRandom, as well as Lambda Invoke. That would be a complete solution for all of those APIs.